LibCarla/source/carla/trafficmanager/LocalizationUtils.cpp

```diff
@@ -232,6 +233,43 @@ namespace LocalizationConstants {
     }
 
   }
+  SimpleWaypointPtr TrackTraffic::GetTargetWaypoint(Buffer& waypoint_buffer,const float& target_point_distance,uint64_t& index)
+  {
+    SimpleWaypointPtr target_waypoint = waypoint_buffer.front();
+    uint64_t  startPosn  = static_cast<uint64_t>(std::fabs(target_point_distance/SAMPLING_RESOLUTION));
+    bool mScanForward = false;
+    double target_point_dist_power = std::pow(target_point_distance, 2);
+
+    /// Condition to determine forward or backward scanning of  WayPoint Buffer.
+    if((startPosn < waypoint_buffer.size()) &&
+        (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+        <= target_point_dist_power)) {
+      mScanForward = true;
+    }
+
+    if(mScanForward) {
+      for (uint64_t j = startPosn ;
+          (j < waypoint_buffer.size()) &&
+          (waypoint_buffer.front()->DistanceSquared(target_waypoint)
```

Comment:
Consider using a reference variable pointing to waypoint_buffer.front() since we will be reusing it many times in the loop and in the else block as well.

Reply:
Agreed and implemented